### PR TITLE
Fixing DerivedShape.getArea() for cold=true

### LINKED ISSUE #1424: `DerivedShape.getArea()` doesn't respect `cold=True` argument
Here is a very simple blueprints file that will run with just the framework:
```yaml
nuclide flags:
  NA:
    burn: false
    xs: true
    expandTo:
  ZR:
    burn: false
    xs: true
    expandTo:
  C:
    burn: false
    xs: true
    expandTo:
  FE:
    burn: false
    xs: true
    expandTo:
  W:
    burn: false
    xs: true
    expandTo:
  SI:
    burn: false
    xs: true
    expandTo:
  CR:
    burn: false
    xs: true
    expandTo:
  MN:
    burn: false
    xs: true
    expandTo:
  NI:
    burn: false
    xs: true
    expandTo:
  V:
    burn: false
    xs: true
    expandTo:
  MO:
    burn: false
    xs: true
    expandTo:
blocks:
    fuel inner: &block_inner_fuel
        fuel:
            shape: Circle
            material: HT9
            Tinput: 25.0
            Thot: 450.0
            id: 0.0
            mult: 271
            od: 0.6029
        duct:
            shape: Hexagon
            material: HT9
            Tinput: 25.0
            Thot: 450.0
            ip: 14.922
            op: 15.710
            mult: 1.0
        intercoolant:
            shape: Hexagon
            material: Sodium
            Tinput: 25
            Thot: 450.0
            ip: duct.op
            op: 16.142
            mult: 1.0
        coolant:
            shape: DerivedShape
            material: Sodium
            Tinput: 25
            Thot: 450.0

assemblies:
    inner fuel:
        specifier: IC
        blocks:
            - *block_inner_fuel
        height: [1,]
        axial mesh points: [1,]
        xs types:
            - A
systems:
    core:
      grid name: core
      origin:
          x: 0.0
          y: 0.0
          z: 0.0
grids:
    core:
        geom: hex
        symmetry: third periodic
        lattice map: |
          IC
```

Run this and then load up the resulting DB to interrogate the component areas:
```python
from armi.bookkeeping.db import databaseFactory
db = databaseFactory("anl-afci-177.h5", 'r')
with db:
     r = db.load(0,0)

for c in r.core[0][0]:
     print(c, c.getArea(cold=True))

<Circle: fuel> 77.3659281298001
<DerivedShape: coolant> 116.6107165005503
<Hexagon: duct> 20.904135052955258
<Hexagon: intercoolant> 11.916564981699693

for c in r.core[0][0]:
     print(c, c.getArea(cold=False))

<Circle: fuel> 78.13119460947178
<DerivedShape: coolant> 116.6107165005503
<Hexagon: duct> 21.1109086835348
<Hexagon: intercoolant> 9.802365546297814
```

Notice that the `DerivedShape` component has the same area no matter if `cold=True` or `cold=False`.

This means that the sum of the component areas changes, even though the `op` of the intercoolant region doesn't change.

```python
sum([c.getArea(cold=True) for c in r.core[0][0]])
226.79734466500537

sum([c.getArea(cold=False) for c in r.core[0][0]])
225.65518533985468
```

In reality, the area of the `DerivedShape` should change as the areas of the other components change.

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Fixed a bug in `DerivedShape.getArea()` in the case when `cold=true`.

## Why is the change being made?

To close #1424

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/components/__init__.py [hunk 1]
```diff
@@ -453,6 +453,15 @@ def getComponentArea(self, cold=False):
         cold : bool, optional
             Ignored for this component
```
Comment (keckler) [inline armi/reactor/components/__init__.py:454]: Please update this docstring

**Review by keckler — APPROVED** [review]
Great, thank you!

File: armi/reactor/components/__init__.py (hunk 1 above)
Comment (john-science) [reply armi/reactor/components/__init__.py]: Fixed

File: armi/reactor/components/__init__.py [hunk 2]
```diff
@@ -158,7 +158,7 @@ def getComponentArea(self, cold=False):
         Parameters
         ----------
         cold : bool, optional
-            Ignored for this component
+            Compute the area with as-input dimensions, instead of thermally-expanded?
```
Comment (keckler) [inline armi/reactor/components/__init__.py:161]: Why the question mark?

Comment (john-science) [reply armi/reactor/components/__init__.py:161]: Oh, do you not like that?  I usually think of booleans as answering a true or false question.

Comment (keckler) [reply armi/reactor/components/__init__.py:161]: Ah, I can read it that way. When I first read it, in my head it just sounded like it was an unsure statement.

I feel like most docstrings in ARMI are not written in that way, but I'm not necessarily anti.

Comment (john-science) [reply armi/reactor/components/__init__.py]: Done

**Comment by keckler** [comment]
Everything looks good by me, thanks @john-science !

**Comment by john-science** [comment]
I have confirmed that this bug fix does not break any downstream unit tests.

**Pull request merged**